### DISCUSSION TITLE
Can Laravel Reverb support Client to Server Web Socket communication?

### DISCUSSION THREAD

**Discussion by trongio** [laravel/framework#51361] [Q&A]
I want to know if it is possible to emit event from client side and Laravel reverb to broadcast it.

I tied many things but couldn't get it to work

**Comment by njzabala** [comment]
In your frontend, use the `send_event` function in Laravel Echo to send your message to laravel. Then, use reverb's `MessageReceived` event listener to broadcast the message.

**Reply by trongio** [reply]
I think I figured it out.
But now there is a problem
I want to differentiate sender and receiver
so player 1 won't receive its own events.
and I think `private channels` work a bit differently.

secondly:
I don't really like this approach
what happens when I want to add another event like this
just add another `if` ?

this is my repo (I want to build a Battleship game 😁):
https://github.com/trongio/Battleship

**Reply by njzabala** [reply]
What I do in this case is I use their Socket Id as Unique Identifier. I don't need to use private channels and no database storage for user account.

**Reply by trongio** [reply]
Can I catch that in listener?

I thought this would work, but no.
```php
broadcast(new SendMessage($data->message))->toOthers();
```

I want my app to work with both users and guests.
generate random `room_id` and use it as an identifier.

won't that need a private channel?

**Reply by njzabala** [reply]
Yes you can get the socket ID connection on the event Listener using `$event->connection->id()`. You can also generate the `room_id` and use it as validation for private channels

**Reply by trongio** [reply]
Thank you I will check it out
